src/api/procedures/setAssetRequirements.ts

```diff
@@ -65,8 +87,10 @@ export async function prepareSetAssetRequirements(
     return !differenceWith(a, b, isEqual).length && a.length === b.length;
   };
 
+  const requirementsCondition = identityTransformation(requirements, context);
```

Comment:
Call the destructured variable `inputRequirements` and the converted one `requirements`